lib/strings.nix

```
@@ -238,7 +247,12 @@ rec {
     # Prefix to check for
     pref:
     # Input string
-    str: substring 0 (stringLength pref) str == pref;
+    str:
+    # Before 23.05, strings would be added to the store before comparing. This was too unexpected.
```

```suggestion
    # Before 23.05, paths would be copied to the store before converting them to strings and comparing. This was surprising and confusing.
```

Also the other instance